### ISSUE TITLE
Documentation: Date input component

### ISSUE THREAD

**Issue by adorayi** [cds-snc/design-gc-conception#699]
**Summary:** Documentation of component guidance in EN for the date input component on the Design System website.
Including use cases, how to use guidance, accessibility guidelines and more.

**Acceptance Criteria:**

- [x] Component Kickoff
- [x] Component review
- [x] Sync with Dev and Design
- [x] Research and consulting other design libraries
- [x] Component ID (name, definition, example image 
- [x] Use case (when to use guidance & related components)  
- [x] How to use guidance 
- [x] Accessibility guidelines
- [x] Inclusive language
- [x] Descriptive text for images
- [x] Translation & Localization
- [x] Bilingual review – content design
- [x] Ensure consistency: ordering, terminology, high level voice/tone/style 
- [x] Final review

**Comment by SmartMouthWords** [comment]
Did an initial review of the [date input documentation ](https://docs.google.com/document/d/17pc3Lg7QYPKtv1xt45GZ0ntX2Tl5oAYRpI-M7bU7hqs/edit), made edits and added questions. 

**Note:** Doc is missing alt text and coding guidance

**Comment by melaniebmn** [comment]
@adorayi as discussed in dev sync: we can't write the coding guidance until we have developed the actual component which is work that was deprioritized a few sprints ago. Can we put this back into the backlog until we have developed the component itself? Dev ticket #698.

**Comment by adorayi** [comment]
@MatildePerrusclet this is ready for review.
https://docs.google.com/document/d/17pc3Lg7QYPKtv1xt45GZ0ntX2Tl5oAYRpI-M7bU7hqs/edit

**Comment by MatildePerrusclet** [comment]
Done reviewing @adorayi @ethanWallace.
Main concern is that there's already guidance regarding the date input order for the GC, it's YYYY-MM-DD for both languages.

Ref:
"Choose a date input order to reduce error
In English, generally use month-day-year and in French day-month-year."

Here are the links to the GC guidance:
https://design.canada.ca/style-guide/#wp4-7
https://www.google.com/url?q=https://www.tbs-sct.canada.ca/pol/doc-eng.aspx?id%3D17284&sa=D&source=docs&ust=1724682711256927&usg=AOvVaw1zuNHNTQeWvIusL-GuHbpU

**Comment by adorayi** [comment]
Good point @MatildePerrusclet. Do we have rationale for why we went with this format? @realdylanzheng @RavinaSamaroo @ethanWallace

**Comment by adorayi** [comment]
Awaiting translation.
@EliseKa please prioritize this.

**Comment by EliseKa** [comment]
Translation is almost complete ([can be found in this doc](https://docs.google.com/document/d/1d2vi_gx3QKO_wP3TjdvwP-KIN0paV04OqyY8e6r2l80/edit)), I'm just awaiting an answer for a clarifying question I asked in the English doc.

**Comment by EliseKa** [comment]
Translation complete and ready for bilingual review - [doc](https://docs.google.com/document/d/1d2vi_gx3QKO_wP3TjdvwP-KIN0paV04OqyY8e6r2l80/edit) already shared with @MatildePerrusclet

**Comment by RavinaSamaroo** [comment]
Anatomy image and overview images were updated.
